### DISCUSSION TITLE
Image Encoding/Decoding

### DISCUSSION THREAD

**Discussion by Cons-Cat** [vlang/v#11902] [Ideas]
I recently started working on a [Kitty graphics protocol](https://sw.kovidgoyal.net/kitty/graphics-protocol/) extension to the term module, and maybe one day Sixel and others could be supported, and a cross-terminal "just works" image rendering function could be written for easy-to-use terminal graphics in V. An issue I ran into is that V doesn't seem to provide any way to reason about graphics formats in its standard library currently, so contributing graphics rendering to the `term` module requires including code unrelated to terminals. For instance, a PNG decoder, which imo shouldn't be in the `term` module because it is generally useful for non-terminal apps as well.

Would a contribution for reading/writing []byte of pixels to/from PNG, JPG, PPM, etc. be welcome? And if so, how would it best be organized? Should there be a module just for image file formats, or should it be a `gg` submodule or something? I saw a way to write PPM in the `x` module (but not read PPM, although that is arguably too trivial). Would that be the best place to keep everything?

**Comment by JalonSolov** [comment]
`x` is for e`x`perimental modules.  Like `json2`, which we hope will one day replace the cJSON stuff.

We do need something like this, and to start it can just be a separate module (outside of V).  If it turns out small, it might be good to put it in vlib, but I have a feeling something like this will be fairly substantial, and vlib is meant to just provide the core routines that almost everyone may need to use, across all different types of programs.

I also think it should be separate from `gg`, since there are a lot of things you could write as command-line only utilities that don't need the rendering... for example, image format conversions (JPG -> PNG, etc.), scaling, etc.
